libs/bragi/src/query.rs

```diff
@@ -482,17 +483,20 @@ fn query(
         .with_source(Source::exclude(&["boundary"]));
 
     // We don't want to clutter the Query URL, so we only add an explanation if the option is used
-    let search_query = match debug {
-        true => search_query.with_explain(true),
-        false => search_query,
+    let search_query = if debug {
+        search_query.with_explain(true)
+    } else {
+        search_query
     };
 
     if let Some(timeout) = &timeout {
         search_query.with_timeout(timeout.as_str());
     }
     let result = search_query.send()?;
 
-    timer.map(|t| t.observe_duration());
+    if let Some(t) = timer {
+        t.observe_duration()
```

Comment:
You can probably add a `;` since the returned value is not used.

```suggestion
        t.observe_duration();
```